dns/client.go

```diff
@@ -102,13 +148,141 @@ func (r *Client) Resolve(
 	return ips, nil
 }
 
-// Lookup resolves a domain name to a slice of IP addresses using the system's
-// default resolver.
+// Lookup resolves a domain name to a slice of IP addresses.
+//
+// It prefers the k6 VU dialer's resolver if available, so that k6 networking
+// options (e.g. blockHostnames) are honored. It falls back to the system's
+// default resolver otherwise.
```

Comment:
By reading this, I feel like we should also call `	r.ensureK6Client()` within this method, but we don't. Is that correct?